main.sh

```diff
@@ -55,68 +55,107 @@ if [ "$1" = "list" ]; then
-
-    if [[ "$use_auth" = "y" || "$use_auth" = "Y" ]]; then
-        read -p " Enter username                 : " username
-        echo -n " Enter password (use %40 for @) : " ; read -s password
+if [ "$1" != "load" ]; then
+    echo -e "
+    \e[1m\e[33mProxyMan
+    =========\e[0m
+    Tool to set up system wide proxy settings on Linux.
+    \e[2m\e[33m🌟\e[0m\e[3m Star it \e[0m : \e[4m\e[34m https://github.com/himanshub16/ProxyMan \e[0m
+
+    \e[4mThe following options are available : \e[0m
+    \e[1m set \e[0m    : \e[2m Set proxy settings \e[0m
+    \e[1m unset \e[0m  : \e[2m Unset proxy settings \e[0m
+    \e[1m list \e[0m   : \e[2m List current settings \e[0m
+
+    "
+
+    read -p " Enter your choice : " choice
+
+    if [[ (! "$choice" = "set") && (! "$choice" = "unset") && (! "$choice" = "list") ]]; then
+        echo "Invalid choice! Start again."
     fi
 
-    if [[ "$use_same" = "y" || "$use_same" = "Y" ]]; then
-        https_host=$http_host
-        ftp_host=$http_host
-        https_port=$http_port
-        ftp_port=$http_port
-        rsync_host=$http_host
-        rsync_port=$https_port
-    else
-        echo -ne "\e[36m HTTPS Proxy host \e[0m " ; read https_host
-        echo -ne "\e[32m HTTPS Proxy port \e[0m " ; read https_port
-        echo -ne "\e[36m FTP   Proxy host \e[0m " ; read ftp_host
-        echo -ne "\e[32m FTP   Proxy port \e[0m " ; read ftp_port
+    if [ "$choice" = "set" ]; then
+
+        echo
+        echo -e " \e[4mEnter details \e[0m : \e[2m\e[3m (leave blank if you don't to use any proxy settings) \e[0m"
+        echo
+        echo -ne "\e[36m HTTP Proxy host \e[0m"; read http_host
+        echo -ne "\e[32m HTTP Proxy port \e[0m"; read http_port
+        echo -ne "\e[0m Use same for HTTPS and FTP (y/n)  ? \e[0m"; read use_same
+        echo -ne "\e[0m Use authentication (y/n)  ? \e[0m        "; read use_auth
+        echo -ne "\e[0m Save settings for later use (y/n) ? \e[0m"; read save_for_reuse
+
+        if [[ "$use_auth" = "y" || "$use_auth" = "Y" ]]; then
+            read -p " Enter username                 : " username
+            echo -n " Enter password (use %40 for @) : " ; read -s password
+        fi
+
+        if [[ "$save_for_reuse" = "y" || "$save_for_reuse" = "Y" ]]; then
+            read -p " Enter config name                 : " profile_name
+        fi
+
+        if [[ "$use_same" = "y" || "$use_same" = "Y" ]]; then
+            https_host=$http_host
+            ftp_host=$http_host
+            https_port=$http_port
+            ftp_port=$http_port
+            rsync_host=$http_host
+            rsync_port=$https_port
+        else
+            echo -ne "\e[36m HTTPS Proxy host \e[0m " ; read https_host
+            echo -ne "\e[32m HTTPS Proxy port \e[0m " ; read https_port
+            echo -ne "\e[36m FTP   Proxy host \e[0m " ; read ftp_host
+            echo -ne "\e[32m FTP   Proxy port \e[0m " ; read ftp_port
+        fi
     fi
 fi
 
+if [ "$1" = "load" ]; then
```

Comment:
This `if` can be placed just above the previous one, which will leave it in proximity to the place where code related to `load` is there. 
Always remember, statements doing the same task should be closer to each other. This helps a lot when the size increases and when debugging.

Moreover, one of the two `if` can be replaced by an `else`. That'd be more elegant when the `if [ $1 == "load" ]; then` is in `if` and the other one as `else`.